---
version: "3.8"

services:
  django:
    networks:
      - traefik-global-proxy

  nginx:
    image: docker.io/nginx
    depends_on:
      - django
    networks:
      - traefik-global-proxy
    expose:
      - "${PROJECT_PORT_EXTERNAL}"
    volumes:
      - ./nginx/conf.d:/etc/nginx/conf.d
      - ./app/staticfiles:/staticfiles:ro
      - ./app/media:/media:ro
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.${PROJECT_NAME}.rule=Host(`${DOMAIN_NAME}`)"
      - "traefik.http.routers.${PROJECT_NAME}.entrypoints=websecure"
      - "traefik.http.routers.${PROJECT_NAME}.tls.certresolver=letsencrypt"

networks:
  traefik-global-proxy:
    external: true
